src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient_private.cs

```diff
@@ -127,56 +106,59 @@ Uri BuildUriForLocksRoute(string key, string label)
             return builder.Uri;
         }
 
-        void BuildBatchQuery(UriBuilder builder, BatchRequestOptions options)
+        void BuildBatchQuery(UriBuilder builder, SettingSelector selector)
         {
-            if (!string.IsNullOrEmpty(options.Key))
-            {
-                builder.AppendQuery(KeyQueryFilter, options.Key);
-            }
-
-            if (!string.IsNullOrEmpty(options.BatchLink))
+            if (selector.Keys.Count > 0)
             {
-                builder.AppendQuery("after", options.BatchLink);
+                var keys = string.Join(",", selector.Keys).ToLower();
+                builder.AppendQuery(KeyQueryFilter, keys);
             }
 
-            if (options.Label != null)
+            if (selector.Labels.Count > 0)
             {
-                if (options.Label == string.Empty)
+                var labelsCopy = new List<string>();
+                foreach (var label in selector.Labels)
                 {
-                    options.Label = "\0";
+                    if (label == string.Empty)
+                    {
+                        labelsCopy.Add("\0");
```

Comment:
Is there a test for this code path?

Reply:
No. I do have issue https://github.com/Azure/azure-sdk-for-net/issues/5384 for that. I need more test to cover filtering in general.

Reply:
Sounds good!